plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -1909,6 +1911,207 @@ read_only::get_table_rows_result read_only::get_table_rows( const read_only::get
+   memcpy(lb_key.data(), prefix.data(), prefix.size());
+   memcpy(lb_key.data() + prefix.size(), lower_bound.data(), lower_bound.size());
+
+   vector<char> ub_key;
+   const string &upper_bound = *p.upper_bound;
+   ub_key.resize(prefix.size() + upper_bound.size());
+   memcpy(ub_key.data(), prefix.data(), prefix.size());
+   memcpy(ub_key.data() + prefix.size(), upper_bound.data(), upper_bound.size());
+
+   // Iterate the range
+   vector<char> row_key;
+   vector<char> row_value;
+
+   auto lb_itr = kv_context.kv_it_create(p.code.to_uint64_t(), prefix.data(), prefix.size());
+   auto ub_itr = kv_context.kv_it_create(p.code.to_uint64_t(), prefix.data(), prefix.size());
+
+   uint32_t lb_key_size = 0;
+   uint32_t lb_value_size = 0;
+   uint32_t lb_key_actual_size = 0;
+   uint32_t ub_key_size = 0;
+   uint32_t ub_value_size = 0;
+   uint32_t ub_key_actual_size = 0;
+
+   // Find lower bound iterator
+   auto status_lb = lb_itr->kv_it_lower_bound(lb_key.data(), lb_key.size(), &lb_key_size, &lb_value_size);
+   EOS_ASSERT(status_lb != chain::kv_it_stat::iterator_erased, chain::contract_table_query_exception,  "Invalid lower bound iterator in ${t} ${i}", ("t", p.table)("i", p.index_name));
+   // Find upper bound iterator
+   auto status_ub = ub_itr->kv_it_lower_bound(ub_key.data(), ub_key.size(), &ub_key_size, &ub_value_size);
+   EOS_ASSERT(status_ub != chain::kv_it_stat::iterator_erased, chain::contract_table_query_exception,  "Invalid upper bound iterator in ${t} ${i}", ("t", p.table)("i", p.index_name));
+
+   lb_key.resize(lb_key_size);
+   status_lb = lb_itr->kv_it_key(0, lb_key.data(), lb_key_size, lb_key_actual_size);
+   ub_key.resize(ub_key_size);
+   status_ub = ub_itr->kv_it_key(0, ub_key.data(), ub_key_size, ub_key_actual_size);
+
+   // iterate the range
+   auto walk_table_row_range = [&]( auto &itr, auto &end_itr, bool reverse ) {
+      auto cur_time = fc::time_point::now();
+      auto end_time = cur_time + fc::microseconds(1000 * 1000);
+
+      uint32_t actual_size = 0;
+      uint32_t key_size;
+      uint32_t val_size;
+      const vector<char> &end_key = (reverse ? lb_key : ub_key);
+      if( reverse ) {
+         key_size = ub_key_size;
+         val_size = ub_value_size;
+      } else {
+         key_size = lb_key_size;
+         val_size = lb_value_size;
+      }
+      auto cmp = itr->kv_it_key_compare(end_key.data(), end_key.size());
+      if( reverse ) cmp *= -1;
+
+      unsigned int count = 0;
+      for( count = 0; cur_time <= end_time && count < p.limit && cmp < 0; cur_time = fc::time_point::now() ) {
+         row_value.clear();
+         row_value.resize(val_size);
+         auto status = itr->kv_it_value(0, row_value.data(), val_size, actual_size);
+
+         fc::variant row_var;
+         if( p.json ) {
+            row_var = abis.binary_to_variant( p.table.to_string(), row_value, abi_serializer::create_yield_function( abi_serializer_max_time ), shorten_abi_errors );
```

Comment:
Same comment as above. Also should probably adjust yield time according to how much time is left instead of just going with `abi_serializer_max_time`